examples/ext-authz/ext-authz-http-service-api.yaml

```diff
@@ -0,0 +1,36 @@
+apiVersion: gateway.kusk.io/v1alpha1
+kind: API
+metadata:
+  name: ext-authz-http-service
+spec:
+  fleet:
+    name: default
+    namespace: default
+  # service name and port should be specified inside x-kusk annotation
+  spec: |
+    openapi: 3.0.0
+    info:
+      title: ext-authz-http-service
+      description: ext-authz-http-service
+      version: '0.0.1'
+    schemes:
+    - http
+    - https
+    x-kusk:
+      auth:
```

Comment:
so will this be added as a securityScheme to the post-processed OpenAPI definition?

Reply:
There were several catches with that. 
1. Auth can be enabled in combination with 1 or more securitySchemes at the same time which would be introducing additional complexity to our setup
2. request header based securitySchemes (basic, apikey, bearer) don't have additional field that would define endpoint and similar
3. We wanted to ship this soon so we opted for the simplest solution

I'd be happy to discuss this further into details as we said this would be the first iteration of auth.

Let us see how this will stick for now.

Reply:
thanks for clarifying @jasmingacic - i think it's a balance between what makes most sense from an API builder/consumer PoV and what is easiest for us - with this approach we should definitely provide a means to get the post-processed OpenAPI definition via an endpoint - otherwise the builder would have to add both our extension **and** the corresponding securitySchema to provide the API consumer with a valid description of the API... 
@crjones thoughts?